cmd/object-handlers.go

```diff
@@ -107,7 +107,12 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 		return
 	}
 
-	objInfo, err := objectAPI.GetObjectInfo(bucket, object)
+	getObjectInfo := objectAPI.GetObjectInfo
+	if api.CacheAPI() != nil {
+		getObjectInfo = api.CacheAPI().GetObjectInfo
```

Comment:
Since `globalCacheObjectAPI` is mutated only at server start (in API functions registration), perhaps there could be a way to set the `getObjectInfo` function in a way that we don't need to check `api.CacheAPI() != nil` on each request?